dns.tf

```diff
@@ -1,7 +1,14 @@
 locals {
-  vpc_ids       = [module.vpc.vpc_id]
-  public_root   = join(".", ["aws", var.network_name, var.namespace, var.root_domain_name])
-  public_domain = join(".", [data.aws_region.current.name, local.public_root])
+  vpc_ids = [module.vpc.vpc_id]
+  # Preserving some old patterns with a little override logic
+  # Useful for geo-routing. Otherwise just leave all domain vars
+  # except for root domain blank
+  subdomain = var.subdomain == "" ? join(".", compact(["aws", var.network_name, var.namespace])) : var.subdomain
+
+  domain = join(".", compact([local.subdomain, var.root_domain_name]))
+
+  public_root   = var.subdomain == "" ? join(".", compact([local.subdomain, var.root_domain_name])) : var.root_domain_name
+  public_domain = var.subdomain == "" ? join(".", [data.aws_region.current.name, local.public_root]) : join(".", [var.subdomain, local.public_root])
```

Comment:
Let's assume:

- var.subdomain (if specified): "blockdata"
- var.root_domain_name: "test.com"
- var.network_name: "kusama"
- var.namespace: "polkadot"
- Our region: "us-east-1"

Effectively, we will have:
_local.subdomain_:

- When var.subdomain is not provided: "aws.kusama.polkadot"
- When it is: "blockdata"

_domain_:

- "aws.kusama.polkadot.test.com"
- or "blockdata.test.com"

_public_root_:

- "aws.kusama.polkadot.test.com"
- or "test.com"

_public_domain_:

- "us-east-1.aws.kusama.polkadot.test.com"
- or "blockdata.test.com"

*I think* this makes sense? This would result in either **api.us-east-1.aws.kusama.polkadot.test.com** or **api.blockdata.test.com**, which seems like what we had before.

Reply:
That was what I was intending.  I'll show you how it works in the CLI but logic flows to fill these fields for the different topologies.